src/components/TablePagination/index.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { UsePaginationInstanceProps, UsePaginationState } from 'react-table';
+import classnames from 'classnames';
+
+import './index.scss';
+
+type ReactTablePaginationProps = {
+  className?: string;
+} & UsePaginationInstanceProps<{}> &
+  UsePaginationState<{}>;
+
+const getVisiblePages = (page: number, total: number) => {
```

Comment:
This function _badly_ needs some commenting and/or named constants, with all the magic numbers it has. As near as I can tell, what it's supposed to do is, given the currently selected `page` and the `total` number of pages, return an array with the visible page numbers. Is this correct? I also don't understand the reasoning behind the different cases.

Reply:
Yeah I can separate out some named variables and comments for clarity.  The visible page numbers at a given time are following USWDS guidelines for pagination.  The logic from this function was taking from existing versions of USWDS pagination.

Reply:
https://designsystem.digital.gov/components/pagination/

Reply:
It'd probably be helpful to link to that page in a comment. I'm also inclined to either export this function or move it into a separate file and export, then add some unit tests for the different cases; everything listed under "Pagination behaviors" on the USWDS page could make for a test case.

Reply:
Update as of https://github.com/CMSgov/easi-app/pull/1425/commits/3b37f6d12509768c5e2a7c9a983f52449399d663 definitely looks better, thanks. Did our UX folks decide to go with 6 slots instead of the 7 that the USWDS page recommends?

Reply:
I was just reviewing the wireframes, and found that it was 6.  I'm looking again, and may review/confirm with UX what the number should be.  If it needs to be 7 I can update that.

Reply:
There are some conflicting principles in the wireframes and USWDS guidelines, mostly dealing with limits on visible pages at all stages of selection.  If it makes sense to hold off on this PR until I sync with UX, that fine.  Otherwise it's a small adjustment, and it still behind a feature flag

Reply:
Let's not hold up this whole PR. I think this is good for now; if you could, create a Jira ticket for syncing with UX about resolving the differences between the wireframes and USWDS guidelines, then we can make changes if necessary.

Reply:
Oh, I just saw https://github.com/CMSgov/easi-app/pull/1425/commits/38326d3237ca0ff33672c906a672c02672b887f1. What were the results of your conversation with UX?

Reply:
I spoke with Natasha, and she shared current pagination wireframes that she is working on in compliance with USWDS.  I have updated the PR in response to the new Figma designs.  The max elements for pagination is 7 (including elipsis).